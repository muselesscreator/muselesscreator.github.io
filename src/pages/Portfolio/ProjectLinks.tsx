import { RiNpmjsFill, RiGithubFill } from 'react-icons/ri';

import ProjectLink from './ProjectLink';
import { ProjectData } from './types';

const ProjectLinks = ({
    githubUrl,
    githubRepo,
    packageName,
    packageUrl,
}: ProjectData) => (
  <div className="project-links">
    <ProjectLink IconComponent={RiGithubFill} href={githubUrl} label={githubRepo} />
    <ProjectLink IconComponent={RiNpmjsFill} href={packageUrl} label={packageName} />
  </div>
);

export default ProjectLinks;

