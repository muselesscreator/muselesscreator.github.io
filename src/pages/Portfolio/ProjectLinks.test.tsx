import {
  vi,
  describe,
  test,
  expect,
} from 'vitest';
import { shallow } from '@muselesscreator/react-shallow-snapshot';
import ProjectLinks from './ProjectLinks';

vi.mock('./ProjectLink', () => ({ default: 'ProjectLink' }));
vi.mock('react-icons/ri', () => ({
  RiNpmjsFill: 'RiNpmjsFillIcon',
  RiGithubFill: 'RiGithubFillIcon',
}));

const project = {
  title: 'Project Title',
  description: 'Project Description',
  category: 'Project Category',
  githubUrl: 'github-url',
  githubRepo: 'github-repo',
  projectUrl: 'project-url',
  packageName: 'package-name',
  packageUrl: 'package-url',
};

describe('ProjectLinks', () => {
  test('snapshot', () => {
    const el = shallow(<ProjectLinks {...project} />);
    expect(el.snapshot).toMatchSnapshot();
    const links = el.instance.findByType('ProjectLink');
    expect(links).toHaveLength(2);
    expect(links[0].props.href).toBe(project.githubUrl);
    expect(links[0].props.label).toBe(project.githubRepo);
    expect(links[1].props.href).toBe(project.packageUrl);
    expect(links[1].props.label).toBe(project.packageName);
  });
});
