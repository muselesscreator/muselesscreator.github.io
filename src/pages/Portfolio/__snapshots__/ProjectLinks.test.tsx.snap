// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProjectLinks > snapshot 1`] = `
<div
  className="project-links"
>
  <ProjectLink
    IconComponent="RiGithubFillIcon"
    href="github-url"
    label="github-repo"
  />
  <ProjectLink
    IconComponent="RiNpmjsFillIcon"
    href="package-url"
    label="package-name"
  />
</div>
`;
